examples/settings/settings_main.c

```diff
@@ -0,0 +1,301 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/mount.h>
+#include <sys/param.h>
+
+#include "system/settings.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * settings_main
+ ****************************************************************************/
+
+int settings_main(int argc, FAR char *argv[])
+{
+  int                  ret;
+  int                  fd;
+  enum settings_type_e stype;
+  char                 text_path[CONFIG_PATH_MAX];
+  char                 bin_path[CONFIG_PATH_MAX];
+  const char           teststr[] = "I'm a string";
+  int                  testval = 0x5aa5;
+  char                 readstr[CONFIG_SYSTEM_SETTINGS_VALUE_SIZE];
+  struct stat          sbuf;
+
+#ifdef CONFIG_EXAMPLES_SETTINGS_USE_TMPFS
+#  ifndef CONFIG_FS_TMPFS
+#    error TMPFS must be enabled
+#  endif
+#  ifndef CONFIG_LIBC_TMPDIR
+#    error LIBC_TMPDIR must be defined
+#  endif
+  if (stat(CONFIG_LIBC_TMPDIR, &sbuf))
+    {
+      ret = nx_mount(NULL, CONFIG_LIBC_TMPDIR, "tmpfs", 0, NULL);
+      if (ret < 0)
+        {
+          printf("ERROR: Failed to mount tmpfs at %s: %d\n",
+                  CONFIG_LIBC_TMPDIR, ret);
+          goto end;
+        }
+    }
+
+  strcpy(bin_path, CONFIG_LIBC_TMPDIR);
+#else
+  strcpy(path, CONFIG_EXAMPLES_SETTINGS_EXISTING_STORAGE);
+  if (path == NULL)
+    {
+      printf("Settings filepath is empty!");
+      goto end;
+    }
+#endif
+
+  strcat(bin_path, "/");
+  strcat(bin_path, CONFIG_EXAMPLES_SETTINGS_FILENAME);
+  strcpy(text_path, bin_path);
+  strcat(bin_path, ".bin");
+  strcat(text_path, ".txt");
```

Comment:
Always use strnxxx for safety reason (to protect against buffer overflow)

Reply:
Is that really necessary here? The length of the string is what it is, and is already limited in length to either the length of the string within the quotes, or CONFIG_PATH_MAX, etc?